src/hotspot/cpu/x86/x86.ad

```diff
@@ -8661,16 +8653,15 @@ instruct vpopcount_integral_reg_evex(vec dst, vec src) %{
 %}
 
 instruct vpopcount_integral_reg_evex_masked(vec dst, vec src, kReg mask) %{
-  predicate((((Matcher::vector_element_basic_type(n->in(1)) == T_INT ||
-               Matcher::vector_element_basic_type(n->in(1)) == T_LONG)) &&
-             VM_Version::supports_avx512_vpopcntdq()) ||
-            (is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
-             VM_Version::supports_avx512_bitalg()));
+  predicate(UsePopCountInstruction &&
+            ((is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
+              VM_Version::supports_avx512_bitalg()) ||
+             (is_non_subword_integral_type(Matcher::vector_element_basic_type(n->in(1))) &&
+              VM_Version::supports_avx512_vpopcntdq())));
```

Comment:
Could be replaced by:
predicate(is_pop_count_instr_target(Matcher::vector_element_basic_type(n->in(1))));
Also no need to check for UsePopCountInstruction any where as it is only meant for scalar popcount.

Reply:
ADLC does not include code written in %source { }  clause .ad file in to generate DFA.cpp
And adding a prediction logic into a new matcher routine or some other files included by dfa.cpp looks like an overkill since its appearing on only two instruction patterns.

Reply:
Thanks,  I already cleaned this up, its used at couple of places and will refactor when I create mainline integration patch.